libyul/optimiser/VarNameCleaner.h

```diff
@@ -0,0 +1,82 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+
+#include <libyul/AsmDataForward.h>
+#include <libyul/optimiser/ASTWalker.h>
+
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace yul
+{
+
+struct Dialect;
+
+/**
+ * Pass to clean variable names with hard-to-read names, that happened due to
+ * disambiguation, such as stripping and normalizing suffixes, if possible.
+ *
+ * Prerequisites: Disambiguator
+ */
+class VarNameCleaner: public ASTModifier
+{
+public:
+	using BlackList = std::vector<std::string>;
+
+	VarNameCleaner(Dialect const& _dialect, BlackList _blacklist = {})
+		: m_dialect{_dialect}, m_blacklist{std::move(_blacklist)}
```

Comment:
I'm not quite sure what to make of the ``_blacklist`` - what's the intended use case? If it won't change for the lifetime of the ``VarNameCleaner``, we could just initialize ``m_usedNames`` to map blacklisted entries to themselves. I'm also not sure whether having a ``std::vector``, sorting it and using ``std::binary_search`` or just having a ``std::set`` of ``std::string``s is better.

Reply:
I agree, should use a set.